From a084a721f2485e4eba81fe50e52fd3f3932024fa Mon Sep 17 00:00:00 2001
From: Jerry Sun <jerry.sun@windriver.com>
Date: Fri, 24 Jan 2020 09:13:40 -0500
Subject: [PATCH 1/1] Wrap Kubernetes host address in square brackets

When constructing the host address string, the address is
not wrapped in square brackets. This does not work in IPv6
Kubernetes deployments. This commit adds square brackets
around the address. IPv4 was also tested to ensure it works
with wrapped address.

Story: 2006711
Task: 38610

Signed-off-by: Jerry Sun <jerry.sun@windriver.com>
---
 storage/kubernetes/client.go | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/storage/kubernetes/client.go b/storage/kubernetes/client.go
index cb7fd28..f16bc77 100644
--- a/storage/kubernetes/client.go
+++ b/storage/kubernetes/client.go
@@ -419,6 +419,9 @@ func inClusterConfig() (cluster k8sapi.Cluster, user k8sapi.AuthInfo, namespace
 		err = fmt.Errorf("unable to load in-cluster configuration, KUBERNETES_SERVICE_HOST and KUBERNETES_SERVICE_PORT must be defined")
 		return
 	}
+	// we need to wrap IPv6 addresses in square brackets
+	// IPv4 also works with square brackets
+        host = "[" + host + "]"
 	cluster = k8sapi.Cluster{
 		Server:               "https://" + host + ":" + port,
 		CertificateAuthority: "/var/run/secrets/kubernetes.io/serviceaccount/ca.crt",
-- 
2.7.4

